haystack/preview/components/embedders/gradient_document_embedder.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from typing import List, Optional, Dict, Any
+
+from haystack.preview import component, Document, default_to_dict
+from haystack.preview.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install gradientai'") as gradientai_import:
+    from gradientai import Gradient
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class GradientDocumentEmbedder:
+    """
+    A component for computing Document embeddings using Gradient AI API..
+    The embedding of each Document is stored in the `embedding` field of the Document.
+
+    ```python
+    p = Pipeline()
+    p.add_component(instance=GradientDocumentEmbedder(), name="document_embedder")
+    p.add_component(instance=DocumentWriter(document_store=InMemoryDocumentStore()), name="document_writer")
+    p.connect("document_embedder", "document_writer")
+    p.run({"document_embedder": {"documents": documents}})
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
+        model_name: str = "bge-large",
+        access_token: Optional[str] = None,
+        workspace_id: Optional[str] = None,
+        host: Optional[str] = None,
+    ) -> None:
+        """
+        Create a GradientDocumentEmbedder component.
+
+        :param model_name: The name of the model to use.
+        :param access_token: The Gradient access token. If not provided it's read from the environment
+                             variable GRADIENT_ACCESS_TOKEN.
+        :param workspace_id: The Gradient workspace ID. If not provided it's read from the environment
+                             variable GRADIENT_WORKSPACE_ID.
+        :param host: The Gradient host. By default it uses https://api.gradient.ai/.
+        """
+        self._host = host
```

Comment:
The first line here should be `gradientai_import.check()`, to make sure the dependencies are in place when you initialize this component.

Reply:
done